### ISSUE TITLE
compute_checksums not working with self hosted S3 solutions

### ISSUE THREAD

**Issue by TaLoN1x** [fluent/fluent-plugin-s3#260]
If I understand correctly compute_checksums true should make fluent to calculate MD5 checksum for object and validate it during S3 upload. I have configuration like this and my object doesnt seem to have MD5 checksum calculated for it, nor I have any errors in logs:

<match test.**>
@type s3

aws_key_id somekey
aws_sec_key someotherkey

s3_bucket test-bucket
check_bucket true
auto_create_bucket true

ssl_verify_peer false
force_path_style true
compute_checksums true
s3_endpoint https://s3.endpoint.local
s3_object_key_format %{path}/${tag[3]}-%H_%{index}.%{file_extension}
path cf/${tag[1]}_${tag[2]}/%Y/%m/%d
time_slice_format %Y%m%d-%H
<buffer tag,time>
@type file
path /opt/buffer/fluent/s3
timekey 3600 # 1 hour partition
timekey_wait 5m
timekey_use_utc false # use utc
chunk_limit_size 5120m

Am I doing something wrong, or this feature is not meant to be utilized like that?

**Comment by repeatedly** [comment]
>  I have configuration like this and my object doesnt seem to have MD5 checksum calculated for it, nor I have any errors in logs:

How to check it?
MD5 calculation is done by aws-sdk-ruby.
S3 stored object doesn't have metadata for this?

**Comment by TaLoN1x** [comment]
Yes, S3 object doesn't have metadata. I've checked with minio and scality S3 solutions

**Comment by repeatedly** [comment]
https://github.com/fluent/fluent-plugin-s3/blob/3c7b89b637c0688eddf50d08a199171287b3650f/lib/fluent/plugin/out_s3.rb#L211
https://github.com/aws/aws-sdk-ruby/blob/40946f6b6b9a6284d8c8b74bc514a11fc5a1f773/gems/aws-sdk-s3/lib/aws-sdk-s3/client.rb#L153

From the code, passing `compute_checksums` parameter seems correct.
Could you check aws-ruby-sdk behaviour without fluentd?
Here is document for aws-sdk-ruby: https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/welcome.html

**Comment by repeatedly** [comment]
This seems aws-sdk-ruby issue. Closed for now.

**Comment by kvr12294** [comment]
@TaLoN1x What happens when the integrity check fails. Does it drop the file that is being uploaded or is there a way it tells you that the file was tampered with on it's way to S3?
